composition_polynomial/src/periodic_columns.rs

```diff
@@ -0,0 +1,199 @@
+use ark_ff::PrimeField;
+use ark_poly::{EvaluationDomain, Polynomial, Radix2EvaluationDomain};
+use fri::lde::MultiplicativeLDE;
+
+#[derive(Debug, Clone)]
+pub struct PeriodicColumn<F: PrimeField> {
+    group_generator: F,
+    column_step: usize,
+    period_in_trace: usize,
+    n_copies: usize,
+    offset_compensation: F,
+    lde_manager: MultiplicativeLDE<F>,
+}
+
+impl<F: PrimeField> PeriodicColumn<F> {
+    pub fn new(
+        values: Vec<F>,
+        group_generator: F,
+        offset: F,
+        coset_size: usize,
+        column_step: usize,
+    ) -> Self {
+        let period_in_trace = values.len() * column_step;
+
+        assert!(period_in_trace != 0);
+        assert!(coset_size % period_in_trace == 0);
+        let n_copies = coset_size / period_in_trace;
+
+        let offset_compensation = offset.pow([n_copies as u64]).inverse().unwrap();
```

Comment:
What's the difference between `start_point` (as used in `get_coset`) and `offset`? Both should be the first evaluation point of the first entry of the first copy of the periodic table?